### PR TITLE
dev: move /dev/* code to separate file

### PR DESCRIPTION
**Pull request by ziemleszcz**
Devices can be enabled/disabled through lwipopts.h file:
	LWIP_ROUTE_DEV (1 by default)
	LWIP_IFSTATUS_DEV (1 by default)
	LWIP_EXT_PF
	LWIP_LINKMONITOR_DEV (0 by default)
/dev/route and /dev/ifstatus internal buffer sizes can be set by:
	LWIP_ROUTE_DEV_BUFFER_SIZE (1024 by default)
	LWIP_IFSTATUS_DEV_BUFFER_SIZE (512 by default)
Linkmonitor device can be set by:
	LWIP_LINKMONITOR_DEV_NAME (en1 by default)

JIRA: DTR-91

## Description

I've moved all code related to devices to a new file as suggested. Devices can be now disabled from "lwipopts.h" and buffer sizes are also configurable. Old configuration is preserved by default. Finally I've decided to move linkmonitor configuration to "lwipopts.h".

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imx6ull).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: port/main.c [hunk 1]
```diff
@@ -577,12 +95,6 @@ int main(int argc, char **argv)
 #endif
 #endif
 
-	mutexCreate(&rt_table.lock);
-
-#if LWIP_EXT_PF
-	init_filters();
```
Comment (anglov) [inline port/main.c:583]: I'm not sure if PF shouldn't be initialised before netif is created – it's create mutex. Not sure what trigger pf to work. 
@nalajcie do you remember any details?

**Review by nalajcie — APPROVED** [review]
LGTM.

@anglov, please update configs in Your projects as needed :).

**Review by anglov — COMMENTED** [review]
Overall LGTM as it improves current code greatly. I see some minor optimisation points (just sleep if no device is created) but I don't think that it gives a lot just because some devs will be needed in every usage case.

My only concern is about PF and initalisation. Maybe @nalajcie knows more if this initialisation  don't have to be done earlier.

File: port/devs.c [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+#include <unistd.h>
+#include <errno.h>
+#include <sys/msg.h>
+#include <sys/threads.h>
+#include <sys/minmax.h>
+#include <posix/utils.h>
+
+
+/* "/dev/route" and "/dev/ifstatus" are available by default */
+#ifndef LWIP_ROUTE_DEV
```
[the remark this answers is not in the document]
Comment (ziemleszcz) [reply port/devs.c:33]: This approach is used by lwip (check "include/lwip/opt.h") - missing options are set to default values.

[the remark this answers is not in the document]
Comment (anglov) [reply port/devs.c:33]: Oh. I see your point. Thanks for explanation. 

Not sure if we shouldn't have our default-opts wrapped with ifdefs and then partiarly update it in projects. And then all defaults have in one opts file.

I agree that your solution is safer that assuming opts to have something.

File: port/main.c (hunk 1 above)
Comment (ziemleszcz) [reply port/main.c:583]: Thanks for pointing this out. Technically PF mutex should be initialized before driver thread is started (PF is called on received packets). It was broken in old code as well.

**Comment by ziemleszcz** [comment]
Found one more issue missed by my honorable reviewers, rt_table.lock mutex was initialized only when LWIP_DEV_ROUTE was enabled.

**Review by anglov — APPROVED** [review]

**Pull request merged**